core/src/test/java/org/apache/calcite/rex/RexNormalizeTest.java

```diff
@@ -0,0 +1,111 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to you under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.calcite.rex;
+
+import org.junit.jupiter.api.Test;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.core.Is.is;
+
+/** Test cases for {@link RexNormalize}. */
+class RexNormalizeTest extends RexProgramTestBase {
+  @Test void digestIsNormalized() {
+    final RexNode node0 = and(
+        or(input(tBool(), 1), input(tBool(), 0)),
+        input(tBool(), 0));
+    final RexNode node1 = and(
+        input(tBool(), 0),
+        or(input(tBool(), 0), input(tBool(), 1)));
+    assertThat(node0.equals(node1), is(true));
+
+    final RexNode node2 = eq(input(tVarchar(), 0), literal("0123456789012345"));
+    final RexNode node3 = eq(literal("0123456789012345"), input(tVarchar(), 0));
+    assertThat(node2.equals(node3), is(true));
```

Comment:
Please refactor this `assertThat(node2.equals(node3), is(true));` into a helper method, so the failure message contains the proper context information (e.g. nodes that are compared with each other)

Reply:
Okey.